compliance_checker/cf/cf.py

```diff
@@ -2590,44 +2590,124 @@ def check_cell_methods(self, ds):
+                # Probably realistically need two dimensions in majority of
+                # practical cases.
+                reasoning.append('The number of dimensions of the climatology variable %s is %s, but the '
+                                 'number of dimensions of the referencing variable %s is %s. The climatology variable '
+                                 'should have %s dimensions' %
+                                 (ds.variables[clim_var.climatology].name,
+                                  ds.variables[clim_var.climatology].ndim,
+                                  clim_var.name,
+                                  clim_var.ndim,
+                                  clim_var.ndim + 1))
+                return ret_val
+            # check that coordinate bounds are in the proper order.
+            # make sure last elements are boundary variable specific dimensions
+            elif (clim_var.dimensions[:] !=
+                  ds.variables[clim_var.climatology].dimensions[:clim_var.ndim]):
+                valid = False
+                reasoning.append(u"Climatology variable coordinates are in improper order: {}. Bounds-specific dimensions should be last".format(
+                                ds.variables[clim_var.climatology].dimensions))
+                return ret_val
+            elif ds.dimensions[ds.variables[clim_var.climatology].dimensions[-1]].size != 2:
+                valid = False
+                reasoning.append(u"Climatology dimension {} should only contain two elements".format(
+                                boundary_variable.dimensions))
+                return ret_val
+
+
+        # otherwise match the following values with for variable with
+        # `cell_methods` attributes
+        # time: method1 within years time: method2 over years
+        # time: method1 within days time: method2 over days
+        # time: method1 within days time: method2 over days time: method3 over years
+        # optionally followed by parentheses for explaining additional
+        # info, e.g.
+        # "time: method1 within years time: method2 over years (sidereal years)"
+
+        meth_regex = "(?:{})".format("|".join(methods))
+        re_string = (r"^time: {0} within (?:years time: {0} over years|"
+                     r"days time: {0} over days"
+                     r"(?: time: {0} over years)?)(?: \([^)]+\))?$".format(meth_regex))
+        # find any variables with a valid climatological cell_methods
+        for cell_method_var in ds.get_variables_by_attributes(cell_methods=lambda s: s is not
+                                                              None):
+            total_climate_count += 1
+            if not re.search(re_string, cell_method_var.cell_methods):
+                reasoning.append('The "time: method within years/days over years/days" format is not correct in variable {}.'.format(cell_method_var.name))
+            else:
+                valid_climate_count += 1
+
+
```

Comment:
Could you reduce the number of blank lines to 1? PEP-8 recommends only a single line in methods and classes, and two lines at the module level.